### PR TITLE
get app running locally

### PR DESCRIPTION
**Pull request by charkour**
Hi @kvlinden and @rpruim. I have some free time this summer, so I'd like to modernize and improve the third-party dependencies for this project so you and the students can more easily add features in the future. If you'd like to talk in more detail, please email me. Thank you

First order of business is to actually run the app locally with modern a modern JS runtime (Node v18)
This fixes bugs to allow the app to run locally with Node v18 and PNPM v8. This updates FullCalendar to v6 to fix some vdom issues. 

To update to Node v18, use [Node Version Manager](https://github.com/nvm-sh/nvm)
To use pnpm, follow their [docs](https://pnpm.io/installation)

Many of the updates were due to ESLint rule changes and auto-formatting.

- [ ] Update Readme

### CODE REVIEW

File: client-course-schedulizer/package.json [hunk 1]
```diff
@@ -12,9 +12,7 @@
     "test": "react-scripts test"
   },
   "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
+    "hooks": {}
```
Comment (charkour) [inline client-course-schedulizer/package.json:19]: Removed husky hook for the time being

File: client-course-schedulizer/.eslintrc.js [hunk 2]
```diff
@@ -7,9 +7,9 @@ module.exports = {
   },
   extends: [
     "react-app",
-    "airbnb",
```
Comment (charkour) [inline client-course-schedulizer/.eslintrc.js:10]: ESLint was causing errors. It's very outdated so for now, we'll work without a full eslint config.

File: client-course-schedulizer/src/utilities/contexts/appContext.ts [hunk 3]
```diff
@@ -8,14 +8,49 @@ export interface AppContext {
...
   state of the app.
 */
 export const AppContext = createContext<AppContext>({
   appDispatch: voidFn,
-  appState: initialAppState,
+  appState: {
+    classes: [],
+    colorBy: 0,
+    constraints: {},
+    departments: [],
```
Comment (charkour) [inline client-course-schedulizer/src/utilities/contexts/appContext.ts:40]: For the time being, this is duplicated.

**Comment by charkour** [comment]
Tests faill because of switching to PNPM

**Comment by kvlinden** [comment]
Thanks for this, Charles. Alas, we've been buried with summer projects and the new MDS program going live next month. We will be fielding a Schedulizer student team in early September, so hopefully we'll have some time to resurrect this system and review your proposed changes.

**Comment by charkour** [comment]
Roger that. I can verify my changes work and merge them in. That way there's no hassle for the students this fall.

**Comment by kvlinden** [comment]
@charkour - I'm finally coming back around to this issue and am having trouble getting the system running locally. I haven't done active development on this system for a while, so I'm probably forgetting some key steps. 

I verified that I've got Node 18.17.1; NPM 10.0.0; PNPM 8.6.0 on WSL2 Linux, and then I tried running the system on both the production and then the fix-local branches.

- npm install worked on production but npm start failed to bring up the system (citing errors in &fullcalendar)
- npm install failed on fix-local (citing errors with eslint).  

(naive?) question: is npm-install and then npm-start the right way to start the local server?

**Comment by charkour** [comment]
> npm install worked on production but npm start failed to bring up the system (citing errors in &fullcalendar)

Yup! I ran into that too. This branch aims to fix those issues.

> npm install failed on fix-local (citing errors with eslint).

The ESLint errors should be fixable, so you'll need to do that quickly. The console will display what needs to be updated.

The correct way to run this is running the commands `pnpm install` then `pnpm start` in the product directory with the `package.json` file.

**Comment by charkour** [comment]
Please feel free to push to this branch. I cannot confidently commit time to dev work, but I'll advise as possible.

**Comment by kvlinden** [comment]
Thanks - I'm not familiar with pnpm. 
It's running now, but generating an error about installing @testing-library/react. I can work on that.

**Pull request closed without merging**